apis/apps/v1alpha1/type.go

```diff
@@ -115,9 +122,77 @@ type LegacyRenderedTemplateSpec struct {
 	ConfigTemplateExtension `json:",inline"`
 }
 
+type ComponentParameters struct {
+	// Specifies the config file name.
+	//
+	// It must be a string of maximum 63 characters, and can only include lowercase alphanumeric characters,
+	// hyphens, and periods.
+	// The name must start and end with an alphanumeric character.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([a-z0-9\.\-]*[a-z0-9])?$`
+	Name string `json:"name"`
+
+	// Specifies the user-defined configuration parameters.
+	//
+	// +optional
+	Parameters map[string]*string `json:"parameters,omitempty"`
+}
+
+type ComponentConfigDescription struct {
+	// Defines the unique identifier of the config file name.
+	//
+	// It must be a string of maximum 63 characters, and can only include lowercase alphanumeric characters,
+	// hyphens, and periods.
+	// The name must start and end with an alphanumeric character.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([a-z0-9\.\-]*[a-z0-9])?$`
+	Name string `json:"name"`
+
+	// Specifies the name of the ParametersDefinition to associate the configuration file.
+	// The ParametersDefinition object defines the format of the configuration file and all parameters type.
+	//
+	// +optional
+	ParametersDef string `json:"parametersDefName,omitempty"`
```

Comment:
ParametersDef should be taken out and decoupled from ConfigTemplate, as some parameters are not present in configuration files.

You can try to place ParametersDef under ComponentDefinition directly.

Reply:
If some parameters reside in Config Templates, they should be identified and referenced in the ParametersDef.

Reply:
What I want to express is that ParameterDef should not be a supplementary description of the Config Template ConfigMap.

Instead, it should be a list of parameters that are allowed to be set for a Component. If modifying one of these parameters requires updating a configuration file, then this configuration file should be referenced in the description of that parameter.